### ISSUE TITLE
`AttributeError: 'NoneType' object has no attribute 'find_library'` when using dependency('vulkan') in C++ projects

### ISSUE THREAD

**Issue by kusma** [mesonbuild/meson#3065]
I'm trying to use the recently-ish added vulkan dependency module (added in #2113), but I get a somewhat odd crash:

```
D:\src\vulkan-test>meson build
The Meson build system
Version: 0.43.0
Source dir: D:\src\vulkan-test
Build dir: D:\src\vulkan-test\build
Build type: native build
Project name: vulkan-test
Native C++ compiler: cl (msvc 18.00.40629)
Build machine cpu family: x86
Build machine cpu: x86
Found Pkg-config: NO
Traceback (most recent call last):
  File "D:\Python34\lib\site-packages\mesonbuild\mesonmain.py", line 352, in run
    app.generate()
  File "D:\Python34\lib\site-packages\mesonbuild\mesonmain.py", line 147, in generate
    self._generate(env)
  File "D:\Python34\lib\site-packages\mesonbuild\mesonmain.py", line 195, in _generate
    intr.run()
  File "D:\Python34\lib\site-packages\mesonbuild\interpreter.py", line 2785, in run
    super().run()
  File "D:\Python34\lib\site-packages\mesonbuild\interpreterbase.py", line 148, in run
    self.evaluate_codeblock(self.ast, start=1)
  File "D:\Python34\lib\site-packages\mesonbuild\interpreterbase.py", line 170, in evaluate_codeblock
    raise e
  File "D:\Python34\lib\site-packages\mesonbuild\interpreterbase.py", line 164, in evaluate_codeblock
    self.evaluate_statement(cur)
  File "D:\Python34\lib\site-packages\mesonbuild\interpreterbase.py", line 177, in evaluate_statement
    return self.assignment(cur)
  File "D:\Python34\lib\site-packages\mesonbuild\interpreterbase.py", line 585, in assignment
    value = self.evaluate_statement(node.value)
  File "D:\Python34\lib\site-packages\mesonbuild\interpreterbase.py", line 175, in evaluate_statement
    return self.function_call(cur)
  File "D:\Python34\lib\site-packages\mesonbuild\interpreterbase.py", line 387, in function_call
    return self.funcs[func_name](node, self.flatten(posargs), kwargs)
  File "D:\Python34\lib\site-packages\mesonbuild\interpreter.py", line 2112, in func_dependency
    dep = dependencies.find_external_dependency(name, self.environment, kwargs)
  File "D:\Python34\lib\site-packages\mesonbuild\dependencies\base.py", line 617, in find_external_dependency
    dep = packages[lname](env, kwargs)
  File "D:\Python34\lib\site-packages\mesonbuild\dependencies\ui.py", line 538, in __init__
    find_lib = self.compiler.find_library(lib_name, environment, lib_path)
AttributeError: 'NoneType' object has no attribute 'find_library'
```

The failing source-line seems to be
```
find_lib = self.compiler.find_library(lib_name, environment, lib_path)
```

**Comment by kusma** [comment]
This only seems to happen if I specify 'cpp' as `list_of_languages` to the `project` function...

**Comment by kusma** [comment]
So, I can work around this by specifying `['cpp', 'c']` instead.

**Comment by kusma** [comment]
Just for reference, here's the minimal `meson.build`-file to reproduce the problem for me:
```meson
project('vulkan-test', 'cpp')
vulkan_dep = dependency('vulkan')
```

**Comment by kusma** [comment]
Just tested on Linux, and I get the same error there. So this seems to be unrelated to Windows.

**Comment by kusma** [comment]
This seems to have been fixed in 133df3b045c8b70b2750fac5961dc85cea2e19b2.
